### ISSUE TITLE
Ch.10 re-play through

### ISSUE THREAD

**Issue by pinheadmz** [saving-satoshi/saving-satoshi#1176]
## https://dev.savingsatoshi.com/en/chapters/chapter-10/intro-2?dev=true

- [ ] #1205

## https://dev.savingsatoshi.com/en/chapters/chapter-10/opening-a-channel-2?dev=true

- [x] Discrepancy here:
> Remember it's the year 2140! A SHA-256 Stout cost 0.00010000 BTC.
> Fill in the two output amounts, 1000 satoshis for Laszlo and the rest for your change.

- [x] #1204

Actually this reveals an issue with the script editor itself. If the initial stack contains something like `SIG(LASZLO)` and the script field is left completely empty, it validates. This is my fault for not being explicit maybe even back in chapter 9 -- we need to write a script where **ONLY** an initial stack of `SIG(LASZLO)` is a valid spend. I think you are already requiring non-empty scripts in the actual challenges so this might be moot. One thing we could do is test every user script with an initial stack of `1` and require validation to fail.

- [x] The "sats" field should be restricted to numbers only, even while typing. I dunno why I am so dumb but I typed the script in that field at first and couldnt figure out what i did wrong

- [x] Its good the "broadcast transaciton" button is disabled but i think itd be helpful to have a tooltip that explains why (we are just going to send the signed tx to Laszlo)
- [ ] saving-satoshi/saving-satoshi-script#12

![Nov-13-2024 10-32-02](https://github.com/user-attachments/assets/f65fcf26-687b-4dd4-8a77-97dbdc469538)

- [x] Another nit, I think "Input 0" should be on top here to line up with "Output 0". This makes some actual protocol sense as well because its like "deposit" is the *name of the coin we are spending in this input*

![Screenshot 2024-11-13 at 10 34 32 AM](https://github.com/user-attachments/assets/b3a79c67-0445-4e41-9235-c55485ec2f50)

## https://dev.savingsatoshi.com/en/chapters/chapter-10/updating-the-state-1?dev=true

- [x] #1188

This may get recursively annoying though -- because NOW, the "refund" tab should show the input 0 (multisig coin) as having only 100,000 sats not 101,000 and then, the refund tx will only be able to send 99,000 sats... so the refund is a full refund minus two miner fees.

- [x] Nit: if you look at the multisig tab the script is mixed case: `OP_2 OP_PUSH PUBKEY(Laszlo) OP_PUSH PUBKEY(You) OP_2 OP_CHECKMULTISIG`

## https://dev.savingsatoshi.com/en/chapters/chapter-10/updating-the-state-3?dev=true
- [x] copy says click "send to laszlo" but that button doesn't actually exist
- [x] saving-satoshi/saving-satoshi#1180
- [x] When you complete this challenge, Laszlo's signature should still say "pending" not ✔️  because he's about to refuse to sign one last time

## https://dev.savingsatoshi.com/en/chapters/chapter-10/updating-the-state-6?dev=true
- [x] I think there is a mix of actual copy for the user, and my "stage directions" for design here. Might need clean up from @satsie but like - in the world of the story does the character have an actual "send to laszlo" button that they click? Maybe we could include a small graphic on the left side showing both signatures with ✅ 
> This time, when you click Send to Laszlo, he smiles and applauds! He signs the child transaction ([X] Laszlo). 

## https://dev.savingsatoshi.com/en/chapters/chapter-10/making-a-payment-1?dev=true
and 
## https://dev.savingsatoshi.com/en/chapters/chapter-10/making-a-payment-2?dev=true
- [x] let's just be consistent with `CODE` style whenever `REVOCATION_KEY_n` appears

## https://dev.savingsatoshi.com/en/chapters/chapter-10/making-a-payment-2?dev=true
- [x] The tab we are working in should just say "commitment" without the "YOU" because we havent introduced the asymmetry concept yet

## https://dev.savingsatoshi.com/en/chapters/chapter-10/making-a-payment-5?dev=true
- [x] typo here: missing a word, should probably be "pay directly to you..."
> The revocable time lock script will use Laszlo's first revocation key PUBKEY(REVOCATION_LASZLO_1) and the large refund output will just directly to you without any fuss.
- [x] #1206

## https://dev.savingsatoshi.com/en/chapters/chapter-10/making-a-payment-6?dev=true
- [x] First copy line shoudl specify which output (the multisig channel open)
> There are several off-chain transactions that spend that output:

- [x] "Commitment 2 (You):" needs to also include output 1: 1000 sats to laszlo
- [x] REPL area says "Lets move on to the first challenge!" [continue] not the first challenge anymore!
- [ ] #1189

## https://dev.savingsatoshi.com/en/chapters/chapter-10/making-a-payment-8?dev=true
I was hoping this challenge would be more about just changing a few numbers, as in:
- [x] #1191
- [x] both outputs of both txs are prefilled with the script and amounts from the *last challenge* https://github.com/saving-satoshi/saving-satoshi/issues/1138
- [x] the "commitment (you)" tab has the wrong text:
```diff
- This is Laszlo’s version of the commitment transaction.
+ This is your version of the commitment transaction.
```
- [x] hints need to be updated because we are starting with Laszlo's transaction not ours
```diff
- • Once you have Laszlo's signature, send him your REVOCATION_YOU_1 
+ • Once you have Laszlo's signature, send him your REVOCATION_YOU_2 
 
Output 0 is spent by EITHER:

-    You, after 700 blocks: SIG(YOU) 1
-    Laszlo: 0 SIG(REVOCATION_YOU_3) SIG(LASZLO) 0 
+   Laszlo, after 700 blocks: SIG(LASZLO) 1
+   You: 0 SIG(REVOCATION_LASZLO_2) SIG(YOU) 0 

- Output 1 is spent by Laszlo SIG(LASZLO)  
+ Output 1 is spent by You SIG(YOU) 
```

I am stuck here because I think the expected tx got flipped around

**Comment by satsie** [comment]
Wow! Thanks for putting this together @pinheadmz! I have made it through the first three sections of chapter 10 (Celebration, The initial funds, and The refund) and I noticed some of the same things you did. Once I'm done playing through I can try to help address some of the items on your list.

If you have the capacity, I have a few open PRs for mostly copy and small design changes to chapter 10. They make a number of updates to the instructions, which you originally wrote and can provide the technical review on:

1. PR for updates to Celebration and The initial funds. Merges my branch into dev: https://github.com/saving-satoshi/saving-satoshi/pull/1174
2. PR for updates to The refund. Merges my branch into my branch for the previous PR (I will change it to target dev if that one gets merged first): https://github.com/satsie/saving-satoshi/pull/1
